### PR TITLE
Automate cert authority

### LINKED ISSUE #9009: Setup automatic certificate authority for HTTPS
Right now, when dev downloads project and runs, HTTPS throws errors about certificates. This is because the self-signed certificates not having a valid signing certificate authority.
If the certificate authority's (CA) certificate is installed and trusted by the computer, it won't throw these errors and HTTPS will work properly.

Cross-platform solution that sets up self-signed certificates so that we don't have HTTPS errors.

https://github.com/daquinoaldo/https-localhost
https://github.com/DylanPiercey/auto-sni
https://www.npmjs.com/package/nodejs-self-signed-certificate-example
https://github.com/cameronhunter/local-ssl-proxy
https://github.com/christianalfoni/create-ssl-certificate - 
image

https://github.com/lukechilds/create-cert
https://github.com/Dexus/pem
https://github.com/smallstep/certificates

### PR DESCRIPTION
**Pull request by ashish-aesthisia**
### <samp>🤖 Generated by Copilot at 3c9070c</samp>

The pull request adds a feature that enables HTTPS support for the Ethereal Engine web app. It modifies the `dev-docker` script to generate and install a self-signed certificate using a custom configuration file (`openssl.cnf`) for the openssl command.

## References
closes #9009 

### <samp>🤖 Generated by Copilot at 3c9070c</samp>

*  Enable HTTPS support for the web app by generating and installing a self-signed certificate ([link](https://github.com/EtherealEngine/etherealengine/pull/9072/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L55-R55), [link](https://github.com/EtherealEngine/etherealengine/pull/9072/files?diff=unified&w=0#diff-885d620f8282b3f2e871abc843cb3883bd136507f4e3c8e2544abc146213f7d7L12-R12), [link](https://github.com/EtherealEngine/etherealengine/pull/9072/files?diff=unified&w=0#diff-46f6e564a1fd3820b0dbe4284649ed53b011ac37e252d37a34e9a80f2e5d2c4bL1-R11))
  * Modify the `dev-docker` script in `package.json` to run the `generate-certs.sh` script after starting the containers ([link](https://github.com/EtherealEngine/etherealengine/pull/9072/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L55-R55))
  * Modify the `generate-certs.sh` script in the `scripts` folder to use a custom configuration file (`openssl.cnf`) for the openssl command ([link](https://github.com/EtherealEngine/etherealengine/pull/9072/files?diff=unified&w=0#diff-885d620f8282b3f2e871abc843cb3883bd136507f4e3c8e2544abc146213f7d7L12-R12))
  * Add the `openssl.cnf` file to the `scripts` folder with the values for the certificate fields ([link](https://github.com/EtherealEngine/etherealengine/pull/9072/files?diff=unified&w=0#diff-46f6e564a1fd3820b0dbe4284649ed53b011ac37e252d37a34e9a80f2e5d2c4bL1-R11))

### <samp>🤖 Generated by Copilot at 3c9070c</samp>

> _`dev-docker` runs_
> _`generate-certs.sh` script_
> _HTTPS in winter_

## QA Steps
_List any additional steps required to QA the changes of this PR, as well as any supplemental images or videos._

## Checklist
- If this PR is still a WIP, convert to a draft
- When this PR is ready, mark it as "Ready for review"
- [ensure all checks pass](https://github.com/etherealengine/etherealengine/wiki/Testing-&-Contributing)
- Changes have been manually QA'd 
- Changes reviewed by at least 2 approved reviewers

### CODE REVIEW

File: scripts/build_minikube.sh [hunk 1]
```diff
@@ -107,6 +107,9 @@ else
   NODE_ENV=$NODE_ENV
 fi
 
+
+./generate-certs.sh
```
Comment (hanzlamateen) [inline scripts/build_minikube.sh:111]: You also need to do this for `scripts/build_microk8s.sh`

File: scripts/generate-certs.sh [hunk 2]
```diff
@@ -9,7 +9,7 @@ case "$(uname -s)" in
      ;;
    Linux)
     mkdir -p ../certs
-    openssl req -nodes -new -x509 -keyout ../certs/key.pem -out ../certs/cert.pem
+    openssl req -nodes -new -x509 -keyout ../certs/key.pem -out ../certs/cert.pem ./opensslcnf
```
Comment (hanzlamateen) [inline scripts/generate-certs.sh:12]: Which domain these certificates are generated for? What about certificate for minio?

https://etherealengine.github.io/etherealengine-docs/docs/host/devops_deployment/microk8s_windows#accept-invalid-certs

**Review by hanzlamateen — CHANGES_REQUESTED** [review]

File: scripts/build_minikube.sh (hunk 1 above)
Comment (ashish-aesthisia) [reply scripts/build_minikube.sh:111]: Added for microk8s

**Comment by hanzlamateen** [comment]
Can you explain, where in our deployment for microk8s/minikube we are specifying the certificates that are generated in `/generate-certs.sh`?

**Comment by ashish-aesthisia** [comment]
> Can you explain, where in our deployment for microk8s/minikube we are specifying the certificates that are generated in `/generate-certs.sh`?

I checked this but there is no separate implementation related to minikube or microk8s for certs. I think they are directly being used from the filesys. As env.default.local also has the paths
Screenshot 2023-10-24 at 7 47 41 AM

**Comment by hanzlamateen** [comment]
> > Can you explain, where in our deployment for microk8s/minikube we are specifying the certificates that are generated in `/generate-certs.sh`?
> 
> I checked this but there is no separate implementation related to minikube or microk8s for certs. I think they are directly being used from the filesys. As env.default.local also has the paths Screenshot 2023-10-24 at 7 47 41 AM

Have you tested this local certificate thing is working fine for minikube & microk8s?

**Comment by hanzlamateen** [comment]
Got this error in CC

```
8:54:31 AM: configure microk8s - ./scripts/build_microk8s.sh: line 120: ./generate-certs.sh: No such file or directory
8:54:31 AM: configure microk8s - "Failed with error code 1."
8:54:31 AM: configure cluster - "Failed with error code 1."
```

[XRE-b86025e1-657e-4f90-8698-3c7dfe7f3b79-2023-10-24T04-04-19.705Z.txt](https://github.com/EtherealEngine/etherealengine/files/13110644/XRE-b86025e1-657e-4f90-8698-3c7dfe7f3b79-2023-10-24T04-04-19.705Z.txt)

**Comment by ashish-aesthisia** [comment]
> Got this error in CC
> 
> ```
> 8:54:31 AM: configure microk8s - ./scripts/build_microk8s.sh: line 120: ./generate-certs.sh: No such file or directory
> 8:54:31 AM: configure microk8s - "Failed with error code 1."
> 8:54:31 AM: configure cluster - "Failed with error code 1."
> ```
> 
> [XRE-b86025e1-657e-4f90-8698-3c7dfe7f3b79-2023-10-24T04-04-19.705Z.txt](https://github.com/EtherealEngine/etherealengine/files/13110644/XRE-b86025e1-657e-4f90-8698-3c7dfe7f3b79-2023-10-24T04-04-19.705Z.txt)

Let me try this once more

**Review by HexaField — COMMENTED** [review]
will this work across all environments? linux, mac, windows?

**Pull request merged**